lib/custom_devise_failure_app.rb

```diff
@@ -9,6 +9,10 @@ def i18n_message(default = nil)
     message.is_a?(Symbol) ? build_message(message) : message.to_s
   end
 
+  def redirect_url
+    request.env["devise_#{warden_message}_failure_redirect_url"] || super
+  end
```

Comment:
The way this works is we assume that a [concurrent session logout throw](https://github.com/18F/identity-idp/blob/61c5e6685f87ee383941d4924ecabd1fab823513/config/initializers/session_limitable.rb#L33) will be handled by Devise's [failure app responder](https://github.com/heartcombo/devise/blob/e2242a95f3bb2e68ec0e9a064238ff7af6429545/lib/devise/failure_app.rb#L42-L50). By default, this would redirect to [the `new_user_session_url`](https://github.com/heartcombo/devise/blob/e2242a95f3bb2e68ec0e9a064238ff7af6429545/lib/devise/failure_app.rb#L140). We're intercepting it to redirect them to the same place they originally intended, with the expectation that the user would be [fully logged out](https://github.com/18F/identity-idp/blob/61c5e6685f87ee383941d4924ecabd1fab823513/config/initializers/session_limitable.rb#L30-L31) at that point. It's not ideal that this incurs an additional redirect hop, but this gives the best assurance that Warden has a `nil` `user` object, since our concurrent session logout behavior only occurs _after_ Warden has already loaded and assigned the user who's being signed out.